go/interpreter/lfvm/gas.go

```diff
@@ -457,26 +457,17 @@ func gasEip2929AccountCheck(c *context, address tosca.Address) error {
 	return nil
 }
 
-func addressInAccessList(c *context) (warmAccess bool, coldCost tosca.Gas, err error) {
-	warmAccess = true
-	if c.isAtLeast(tosca.R09_Berlin) {
-		addr := tosca.Address(c.stack.peekN(1).Bytes20())
+func calculateAccessCost(address tosca.Address, revision tosca.Revision, runContext tosca.RunContext) (isWarm bool, accessCost tosca.Gas) {
+	isWarm = true
+	if revision >= tosca.R09_Berlin {
 		// Check slot presence in the access list
 		//lint:ignore SA1019 deprecated functions to be migrated in #616
```

Comment:
Since we are already at it, let's cover this TODO and get rid of the deprecated function.